### DISCUSSION TITLE
aligned deallocation function [...] is only available on macOS 10.14 or newer

### DISCUSSION THREAD

**Discussion by nschloe** [wjakob/nanobind#312] [Q&A]
I'm trying to build a nanobind project on GitHub's macos runners. My `pyproject.toml` includes
```toml
[tool.cibuildwheel.macos.environment]
MACOSX_DEPLOYMENT_TARGET = "10.14"
```
and the runner reports
```
Operating System
  macOS
  12.7
  21G816
```
Still, I'm getting
```
    /path/to/overlay/lib/python3.8/site-packages/nanobind/src/nb_type.cpp:207:13: error: aligned deallocation function of type 'void (void *, std::align_val_t) noexcept' is only available on macOS 10.14 or newer
                operator delete(p, std::align_val_t(t->align));
                ^
```
Any idea what might be going wrong?

_Edit:_

Somehow it falls back to macosx 10.9:
```
Download https://www.python.org/ftp/python/3.8.10/python-3.8.10-macosx10.9.pkg to [...]
```
Investigating...

Ref.: https://github.com/pypa/cibuildwheel/discussions/1636

**Comment by wjakob** [comment]
Not sure, but in general aligned versions of these are a C++17 feature, which is in turn a strict nanobind requirement.

**Comment by nschloe** [comment]
The culprit was a 
```
CIBW_ENVIRONMENT
```
setting in the github actions yaml. This apparently removes all other env settings, including the ones `[tool.cibuildwheel.macos.environment]`. I've now put all my config in the yaml, and it works.
